lib/temple/utils.rb

```diff
@@ -18,6 +18,8 @@ module Utils
     # @param html [String] The string to escape
     # @return [String] The escaped string
     def escape_html_safe(html)
```

Comment:
Something strange is happening here. The documentation says that `html` must already be a String. There seems to be a case here where it's *not* getting a string. Do you know why the caller doesn't call `#to_s`?

Reply:
Ah, I see.  I thought it was a feature, since Temple defines `Object#html_safe?`, and that isn't called anywhere else.  And also because `escape_html` does accept non-Strings (i.e. calls `to_s`).  I will investigate further.

Reply:
I believe the reason `escape_html_safe` is receiving a non-String is because `Temple::Filters::Escapable` does not include a call to `to_s`:

https://github.com/judofyr/temple/blob/7987ab67af00a598eb3d83192415371498a0f125/lib/temple/filters/escapable.rb#L17-L18

`@escape_code` is used as-is in `Temple::Filters::Escapable#on_dynamic`:

https://github.com/judofyr/temple/blob/7987ab67af00a598eb3d83192415371498a0f125/lib/temple/filters/escapable.rb#L35-L37

And Slim triggers `on_dynamic`:

https://github.com/slim-template/slim/blob/e0908c9fffde402610ac19fe96bbc1e9484896c6/lib/slim/controls.rb#L47

Additionally, I think the issue is not seen when using pretty because `Temple::HTML::Pretty` evaluates the code it receives, turning the result into a String.  I haven't dug into it as deeply, but I think that's happening here:

https://github.com/judofyr/temple/blob/7987ab67af00a598eb3d83192415371498a0f125/lib/temple/html/pretty.rb#L36-L40

And you can see that `Pretty` occurs before `Escape` when using Slim:

https://github.com/slim-template/slim/blob/6e74e9b1e5775cf15a1f90a2abae1309d30d8dd2/lib/slim/engine.rb#L32-L33

Reply:
With all that said, I am still preferential to the solution in this PR.  I can amend the method docs for `escape_html_safe` (and `escape_html`) if you'd like.

Alternatively, I can add a `to_s` to `@escape_code`, as such:

```ruby
 @escape_code = options[:escape_code] || 
   "::Temple::Utils.escape_html#{options[:use_html_safe] ? '_safe' : ''}((%s).to_s)" 
```